research/mark_generator/training_data_generator.py

```diff
@@ -0,0 +1,77 @@
+import random
+import typing as tp
+from pathlib import Path
+
+import cv2
+import numpy as np
+
+from research.data_augmentator import DataAugmentator
+from research.mark_generator import MarkGenerator, Mark
+
+
+class TrainingDataGenerator:
+    ANSWER_BOXES_PATH = 'data/input/answer_boxes.jpg'
+    INDEX_BOXES_PATH = 'data/input/index_boxes.jpg'
+    OUTPUT_DIR = 'data/output'
+
+    def __init__(self, shape: tp.Tuple = (90, 90), box_offset: int = 4):
+        self._border = 5
+        self._shape = shape
+        self._box_offset = box_offset
+        self._raw_answer_boxes = self.extract_boxes()
+        self._mark_generator = self.set_mark_generator()
+
+    def set_mark_generator(self):
+        mark, unmark = Mark.get_random_valid_marks()
+        alpha = random.uniform(0.05, 0.2)
+        beta = random.uniform(0.05, 0.2)
+        gamma = random.uniform(0.05, 0.2)
+        rho = random.uniform(0.05, 0.2)
+        weight = (random.randint(3, 8), random.randint(3, 8))
+        return MarkGenerator(self._shape, mark, unmark, alpha, beta, gamma, rho, weight)
+
+    def generator(self, idx: int = 0, augmentation: bool = True):
+        Path(self.OUTPUT_DIR).mkdir(parents=True, exist_ok=True)
+        while True:
+            marked_box, label = self.create_marked_box()
+            if augmentation:
+                marked_box = DataAugmentator(marked_box).gaussian_noise().shift().get()
```

Comment:
It would be great if we have some random chance for not marking any box (returning empty mark box)

Reply:
It's already done inside create_marked_box function